### ISSUE TITLE
API change idea

### ISSUE THREAD

**Issue by SamVerschueren** [Qix-/node-simple-swizzle#1]
Hi @Qix- 

I like the module, will definitely use it in some of my projects. I saw you using it in [color-string](https://github.com/Qix-/color-string/pull/22) and it gave me an idea. It's just something that came in my mind and thought I could share. If you don't like it, fine :).

So you always have to call `swizzle(arguments)` and assign that to a variable.

``` js
cs.to.hsl = function () {
    var hsla = swizzle(arguments);
    return hsla.length < 4 || hsla[3] === 1
        ? 'hsl(' + hsla[0] + ', ' + hsla[1] + '%, ' + hsla[2] + '%)'
        : 'hsla(' + hsla[0] + ', ' + hsla[1] + '%, ' + hsla[2] + '%, ' + hsla[3] + ')';
};
```

Nothing wrong with that, but it could be done like this

``` js
cs.to.hsl = swizzle(function (hsla) {
    return hsla.length < 4 || hsla[3] === 1
        ? 'hsl(' + hsla[0] + ', ' + hsla[1] + '%, ' + hsla[2] + '%)'
        : 'hsla(' + hsla[0] + ', ' + hsla[1] + '%, ' + hsla[2] + '%, ' + hsla[3] + ')';
});
```

The user example in the `readme` will then look like this

``` js
var swizzle = require('simple-swizzle');

var myFunc = swizzle(function (args) {
    return args;
});

myFunc(1, [2, 3], 4); // [1, 2, 3, 4]
myFunc(1, 2, 3, 4);   // [1, 2, 3, 4]
myFunc([1, 2, 3, 4]); // [1, 2, 3, 4]
```

Just something I couldn't resist sharing :).

**Comment by Qix-** [comment]
Definitely, though I wonder if a magical parameter (function|`arguments`) would be confusing. What about a function tacked on to the end, like `swizzle.wrap()`?

**Comment by SamVerschueren** [comment]
Not sure if I understand correctly. Mind to give an example?

**Comment by Qix-** [comment]
``` js
var swizzle = require('simple-swizzle');

function foo() {
    var args = swizzle(arguments); // ok
}

var foo = swizzle(function (args) {
}); // a bit too magic for the same overload?

// how about...

function foo() {
    var args = swizzle(arguments);
}

var foo = swizzle.wrap(function (args) {
    // at least the `swizzle.wrap()` is explicit.
});
```

Just a thought. It'd cut down on branching within the function too, meaning it won't have to check if the argument passed to `swizzle()` is a function (wrap) or an arguments object (perform swizzling).

Though I do think this module _should_ be able to do this, definitely.

**Comment by SamVerschueren** [comment]
You're right. With using wrap, you can still swizzle function as well. Not sure if you want that, but to give an example

``` js
function foo() {
    var fnArray = swizzle(arguments);
}

function handler1() {}
function handler2() {}

foo(handler1, handler2);
```

With using `swizzle.wrap`, you can still do that because it is explicit. :+1:

**Comment by Qix-** [comment]
Yep, exactly. I'll add that in really quick. :+1: thanks for the idea!

**Comment by Qix-** [comment]
Landed in 366bf067f1c7fe84b20bd8cffdb6eedf028d3be4 and published as `0.2.0` :+1: Thanks!!

**Comment by SamVerschueren** [comment]
That was fast :). Thanks!
